zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterEventPoller.java

```diff
@@ -72,6 +72,8 @@ public void setInterpreterGroup(InterpreterGroup interpreterGroup) {
   @Override
   public void run() {
     Client client = null;
+    AppendOutputRunner.setListener(listener);
```

Comment:
Changed the class AppendOutputRunner to non-static. It has made code and tests simpler too. This is ready for review now.